ci/ci/github.py

```diff
@@ -928,14 +948,17 @@ async def _heal(self, db: Database, batch_client: BatchClient, gh: gh_aiohttp.Gi
         for pr in self.prs.values():
             # merge candidate if up-to-date build passing, or
             # pending but haven't failed
-            if pr.review_state == 'approved' and not pr.build_failed_on_at_least_one_platform():
-                pri = pr.merge_priority()
-                is_authorized = await pr.authorized(db)
-                if is_authorized and (
-                    not merge_candidate or (merge_candidate_pri is not None and pri > merge_candidate_pri)
-                ):
-                    merge_candidate = pr
-                    merge_candidate_pri = pri
+            if pr.review_state == 'approved':
+                if not pr.build_failed_on_at_least_one_platform():
+                    pri = pr.merge_priority()
+                    is_authorized = await pr.authorized(db)
+                    if is_authorized and (
+                        not merge_candidate or (merge_candidate_pri is not None and pri > merge_candidate_pri)
+                    ):
+                        merge_candidate = pr
+                        merge_candidate_pri = pri
+                else:
+                    log.info(f'{pr.short_str()} is not a merge candidate because it has failed on at least one platform')
```

Comment:
```suggestion
                    log.info(f'{pr.short_str()} is not a merge candidate because it has failed on the following platform(s): {[platform for platform, status in self.last_known_github_status.items() if status == GithubStatus.FAILURE]}')
```